app/ante/max_tx_size.go

```diff
@@ -0,0 +1,29 @@
+package ante
+
+import (
+	"github.com/celestiaorg/celestia-app/v3/pkg/appconsts"
+	v3 "github.com/celestiaorg/celestia-app/v3/pkg/appconsts/v3"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// MaxTxSizeDecorator ensures that a tx can not be larger than
+// application's configured versioned constant.
+type MaxTxSizeDecorator struct{}
+
+func NewMaxTxSizeDecorator() MaxTxSizeDecorator {
+	return MaxTxSizeDecorator{}
+}
+
+// AnteHandle implements the AnteHandler interface. It ensures that tx size is under application's configured threshold.
+func (d MaxTxSizeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	// Tx size rule applies to app versions v3 and onwards.
+	if ctx.BlockHeader().Version.App < v3.Version {
+		return next(ctx, tx, simulate)
+	}
+
+	if len(ctx.TxBytes()) > appconsts.MaxTxBytes(ctx.BlockHeader().Version.App) {
+		return ctx, sdkerrors.ErrTxTooLarge
```

Comment:
[nit] The current error lacks context for a user to understand how large their transaction is and how large of a tx the system can support.

Proposal to include two more pieces of information in this error: the size of the current tx and the max tx bytes.

Reply:
Also for posterity it looks like `mempool.ErrTxTooLarge` is thrown by Tendermint if the tx exceeds the mempool's max tx bytes parameter.

In BroadcastTx the Cosmos SDK translates the `mempool.ErrTxTooLarge` into a `sdkerrors.ErrTxTooLarge` which is the same error used here.

[proposal] I wonder if we should use a different error to differentiate that this error is _not_ because the tx exceeds the mempool's configured max tx bytes but instead is the application's max tx bytes. Since the mempool can be configured locally, user's may or may not observe a `mempool.ErrTxTooLarge` (and therefore `sdkerrors.ErrTxTooLarge` depending on which node they submit transactions to.

Reply:
I can address these in a follow-up since I need to link the new versioned constant in ante handler docs anyway